### PR TITLE
Like functionality for post

### PR DESCRIPTION
**Pull request by ddrmv**
Adding Like functionality with maximum of 1 like per user, using array of strings in the post model. Added a require for hbs, so we can add handlebars helpers. Added a handlebar helper function in app.js to show "1 like" or "x likes" in the like counter. Maybe we can move the handlebar helper functions to an external file later?
Accidentally deleted the div post from the post index handlebar and re-added it in a later commit.
Not sure about the class names for the form elements in views/posts/index.hbs - maybe should be more descriptive, e.g. likeSubmit, etc?
Possible further improvements: hide or grey out the Like button when already liked that specific post; maybe not be able to like the post if it is by the same user.

### CODE REVIEW

**Comment by ddrmv** [comment]
[(L) implement like button and like count](https://trello.com/c/LTJUw3ek/34-l-implement-like-button-and-like-count)

**Comment by ddrmv** [comment]
[change like from number to array which stores user info who liked the post](https://trello.com/c/fBxlxXb9/37-change-like-from-number-to-array-which-stores-user-info-who-liked-the-post)

**Comment by ddrmv** [comment]
[(M) The user ID is stored in the like count](https://trello.com/c/lUfMZvNz/40-m-the-user-id-is-stored-in-the-like-count)

File: views/posts/index.hbs [hunk 1]
```diff
@@ -5,6 +5,12 @@
       <li>{{this.author.name}}</li>
       <li>{{this.createdAt}}</li>
       <li>{{this.message}}</li>
+      <li class="likeCount">{{formatLikesNumber this.like.length}}</li>
+      <form class="like" action="/posts/like" method="post">
+        <input class="postid" type="hidden" value={{this._id}} name="postid" />
+        <input class="like" type="hidden" value={{this.like}} name="like" />
```
Comment (ritaaktay) [inline views/posts/index.hbs]: No need for `this.like` to be sent as `body.like`

File: controllers/posts.js [hunk 2]
```diff
@@ -28,6 +28,18 @@ const PostsController = {
...
   },
+  Like: (req, res) => {
+    Post.updateOne(
+      { _id: req.body.postid },
+      { $addToSet: { like: req.session.user._id } }
+    ).exec((err, post) => {
+      if (err) {
+        throw err;
+      }
+      console.log(post.like);
```
Comment (ritaaktay) [inline controllers/posts.js]: we can comment this out (console.log)

File: cypress/integration/user_can_like_a_post_once_spec.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    cy.get("#name").type("someone4");
+    cy.get("#email").type("someone4@example.com");
+    cy.get("#password").type("password");
+    cy.get("#submit").click();
+
+    cy.visit("/users/new");
+    cy.get("#name").type("someone5");
+    cy.get("#email").type("someone5@example.com");
+    cy.get("#password").type("password");
+    cy.get("#submit").click();
```
Comment (ritaaktay) [inline cypress/integration/user_can_like_a_post_once_spec.js:14]: Cross-check emails with Jovi's tests

**Pull request merged**